app/models/task.py

```diff
@@ -3,3 +3,38 @@
 
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable = True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+    goal = db.relationship("Goal", back_populates="tasks")
+
+
+    def to_dict(self):
+        
+        if self.goal_id:
+            return {
+            "id": self.task_id,
+            "goal_id":self.goal_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_complete()
+        }
+            
+        else:
+            return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_complete()
+        }
```

Comment:
We can refactor this `to_dict` by setting a variable to our dictionary and then adding a check for whether a goal_id exists:

```python
    def to_dict(self):
        task_dict = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.is_complete()
        }
            
        if self.goal_id:
            task_dict["goal_id"] = self.goal_id
            
        return task_dict
```